lucet-validate/src/main.rs

```diff
@@ -87,14 +87,14 @@ fn run(module_path: &Path, witx_path: &Path, wasi_exe: bool) -> Result<(), Error
     Ok(())
 }
 
-#[derive(Debug, Fail)]
+#[derive(Debug, Error)]
 enum Error {
-    #[fail(display = "{}", _0)]
-    Witx(#[cause] witx::WitxError),
-    #[fail(display = "With file {:?}: {}", _0, _1)]
-    Io(PathBuf, #[cause] io::Error),
-    #[fail(display = "{}", _0)]
-    Validate(#[cause] lucet_validate::Error),
+    #[error("{}", _0)]
+    Witx(#[source] witx::WitxError),
```

Comment:
Some opportunities for `#[from]` on the witx and validate variants

Reply:
Hm.  Thanks so much for this comment, but I'm having a bit of trouble reading it.  Is this a comment on changes that could be made over in `witx` or changes that could be made in the scope of this PR?  I look forward to learning more.

Reply:
Just in this file. You should be able to replace the explicit `From` impls below this by adding the `#[from]` annotation next to the field that they'd otherwise be turned into. I don't believe `#[from]` was a feature available to us in `failure`; yay for less boilerplate!